local/compose.go

```diff
@@ -65,6 +70,100 @@ func (s *composeService) Build(ctx context.Context, project *types.Project) erro
 	return s.build(ctx, project, opts)
 }
 
+func (s *composeService) Push(ctx context.Context, project *types.Project) error {
+	configFile, err := config.Load(config.Dir())
+	if err != nil {
+		return err
+	}
+	eg, ctx := errgroup.WithContext(ctx)
```

Comment:
```suggestion
	eg, _ := errgroup.WithContext(ctx)
```

The `ctx` returned by the `WithContext` doesn't end on ctrl+c

Reply:
ack. But then I wonder: an error in one of the goroutines will have errgroup to invoke it's `cancel` to stop others. If we don't use this context we will have to wait for all of them to complete. How to get both?

Reply:
Looking at `WithCancel` source it seems it has been designed to propagate cancel events from parent, so this _should_ work. But maybe I missunderstand how to use this

Reply:
just confirmed [ctrl+c] works with context returned by `errgroup.WithContext`